homeassistant/components/renson/binary_sensor.py

```diff
@@ -0,0 +1,138 @@
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator
+from .const import DOMAIN
+from .entity import RensonEntity
+
+
+@dataclass
+class RensonBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    field: FieldEnum
+
+
+@dataclass
+class RensonBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RensonBinarySensorEntityDescriptionMixin
+):
+    """Description of binary sensor."""
+
+
+BINARY_SENSORS: tuple[RensonBinarySensorEntityDescription, ...] = (
+    RensonBinarySensorEntityDescription(
+        translation_key="frost_protection_active",
+        key="FROST_PROTECTION_FIELD",
+        field=FROST_PROTECTION_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_ENABLE_FIELD",
+        translation_key="breeze",
+        field=BREEZE_ENABLE_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_MET_FIELD",
+        translation_key="breeze_conditions_met",
+        field=BREEZE_MET_FIELD,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="HUMIDITY_CONTROL_FIELD",
+        translation_key="humidity_control",
+        field=HUMIDITY_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="AIR_QUALITY_CONTROL_FIELD",
+        translation_key="air_quality_control",
+        field=AIR_QUALITY_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="CO2_CONTROL_FIELD",
+        translation_key="co2_control",
+        field=CO2_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="PREHEATER_FIELD",
+        translation_key="preheater",
+        field=PREHEATER_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Call the Renson integration to setup."""
+
+    api: RensonVentilation = hass.data[DOMAIN][config_entry.entry_id].api
+    coordinator: RensonCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ].coordinator
+
+    entities: list = []
+    for description in BINARY_SENSORS:
+        entities.append(RensonBinarySensor(description, api, coordinator))
+
+    async_add_entities(entities)
+
+
+class RensonBinarySensor(RensonEntity, BinarySensorEntity):
+    """Get sensor data from the Renson API and store it in the state of the class."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
```

Comment:
None of these devices are the main device and they get their names from the entity description so this should be removed